src/bft.py

```diff
@@ -28,5 +30,13 @@ def bf_order(t: T | None) -> Iterable[int]:
     L = []
     if(t!=None):
         D.append(t)
-        bf_order_D_L(D, L)
+        while D!=deque([]):
```

Comment:
You can just use `while D` to check if `D` is non-empty. Creating a whole new `deque` just to check for emptiness is rather wasteful in time.